tough/src/schema/mod.rs

```diff
@@ -324,6 +392,214 @@ impl Role for Targets {
     }
 }
 
+//Implementation for delegated targets
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
+pub struct Delegations {
+    #[serde(deserialize_with = "de::deserialize_keys")]
+    pub keys: HashMap<Decoded<Hex>, Key>,
+    pub roles: Vec<DelegatedRole>,
+}
+
+///Each role delegated in a targets file is considered a delegated role
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
+pub struct DelegatedRole {
+    pub name: String,
+    pub keyids: Vec<Decoded<Hex>>,
+    pub threshold: NonZeroU64,
+    #[serde(flatten)]
+    paths: PathSet,
+    terminating: bool,
+    #[serde(skip)]
+    pub targets: Option<Signed<Targets>>,
+}
+
+///Targets can delegate paths as paths or path hash prefixes
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
+pub enum PathSet {
+    #[serde(rename = "paths")]
+    Paths(Vec<String>),
+
+    #[serde(rename = "path_hash_prefixes")]
+    PathHashPrefixes(Vec<String>),
+}
+
+impl PathSet {
+    ///Given a target string determines if paths match
+    fn matched_target(&self, target: &str) -> bool {
+        match self {
+            Self::Paths(paths) => {
+                for path in paths {
+                    if Self::matched_path(path, target) {
+                        return true;
+                    }
+                }
+            }
+
+            Self::PathHashPrefixes(path_prefixes) => {
+                for path in path_prefixes {
+                    if Self::matched_prefix(path, target) {
+                        return true;
+                    }
+                }
+            }
+        }
+        false
+    }
+
+    ///Given a path hash prefix and a target path determines if target is delegated by prefix
+    fn matched_prefix(prefix: &str, target: &str) -> bool {
+        let temp_target = target.to_string();
+        let hash = digest(&SHA256, temp_target.as_bytes());
+        hash.as_ref().starts_with(prefix.as_bytes())
+    }
+
+    ///Given a shell style wildcard path determines if target matches the path
+    fn matched_path(wildcardpath: &str, target: &str) -> bool {
+        let mut regex_string = wildcardpath.to_string();
+        regex_string = regex_string.replace(".", "\\.");
+        regex_string = regex_string.replace("*", "[^/]*");
+        regex_string = regex_string.replace("?", ".");
+        let re = Regex::new(&regex_string);
+        if let Ok(re) = re {
+            re.is_match(&target)
+        } else {
+            false
+        }
+    }
+}
```

Comment:
`wildcardpath` might contain other characters which are special in regexes, we would need to escape those.

Because the spec says `PATHPATTERN can include shell-style wildcards and supports the Unix filename pattern matching convention`, it might be more straightforward to use a higher-level matching library like [globset](https://docs.rs/globset), especially since it efficiently matches over a list of globs. globset's support of Unix-style glob syntax is wider than listed in the TUF spec's examples, but it's impossible to read between the lines of specification vagueness.